packages/uswds-core/src/js/utils/focus-trap.js

```diff
@@ -3,42 +3,61 @@ const behavior = require("./behavior");
 const select = require("./select");
 const activeElement = require("./active-element");
 
+console.log("hello");
+
 const FOCUSABLE =
   'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, [tabindex="0"], [contenteditable]';
 
 const tabHandler = (context) => {
-  const focusableElements = select(FOCUSABLE, context);
-  const firstTabStop = focusableElements[0];
-  const lastTabStop = focusableElements[focusableElements.length - 1];
+  let focusableElements = [];
+  
+  // Creates a new mutationObserver to watch for changes
+  // Observers callback function that updates the focusable elements when change is detected
+  const observer = new MutationObserver(() => {
+    updateFocusableElements();
+  });
+
+  function updateFocusableElements() {
+    focusableElements = Array.from(select(FOCUSABLE, context));
+  }
+  
+  const firstTabStop = () => focusableElements[0];
+  const lastTabStop = () => focusableElements[focusableElements.length - 1];
```

Comment:
Rather than attaching a mutation observer, would it be simpler to get the focusable elements on demand when handling tab changes?

```suggestion
  const getFocusableElements = () => select(FOCUSABLE, context);
  const firstTabStop = () => getFocusableElements()[0];
  const lastTabStop = () => getFocusableElements().at(-1);
```

Reply:
@aduth Thank you for this recommendation! I agree, it would be a simpler fix than using the mutation observer. I've adjusted the code to implement your recommendations, with a few changes.